pkg/api/v1alpha1/cluster_types.go

```diff
@@ -1227,14 +1249,23 @@ func (c *Cluster) ConvertConfigToConfigGenerateStruct() *ClusterGenerate {
 	return config
 }
 
+// IsManaged returns true if the Cluster is not self managed.
 func (c *Cluster) IsManaged() bool {
 	return !c.IsSelfManaged()
 }
 
+// ManagedBy returns the Cluster's management cluster's name.
 func (c *Cluster) ManagedBy() string {
 	return c.Spec.ManagementCluster.Name
 }
 
+// EKSAReleaseName maps EksaVersion to an EKSARelease.
+func (c *Cluster) EKSAReleaseName() string {
+	name := "eksa-"
+	version := string(*c.Spec.EksaVersion)
+	return name + strings.ReplaceAll(version, ".", "-")
```

Comment:
```suggestion
	return fmt.Sprintf("eksa-%s", strings.ReplaceAll(version, ".", "-")
```

Reply:
also do we need to validate the format of version here if not taken care of in somewhere else

Reply:
version would be validated before this is called